pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/ConfigRunner.cs

```diff
@@ -200,6 +241,92 @@ public void WriteBatchCommandsToFile(CommandWriter commandWriter, int startStep)
             }
         }
 
+        private async Task DownloadData()
+        {
+            var mainSettings = Config.MainSettings;
+            var server = mainSettings.Server;
+            if (!server.Validated)
+                server.Validate();
+
+            var allFiles = server.GetServerFiles;
+            var fileNames = allFiles.Keys;
+            var existingDataFiles = Directory.GetFiles(mainSettings.DataFolderPath).Length;
+            var dataFilter = new Regex(mainSettings.Server.DataNamingPattern);
+            var downloadingFilesEnum =
+                from name in fileNames
+                where dataFilter.IsMatch(name)
+                select name;
+            var downloadingFiles = downloadingFilesEnum.ToList();
+            if (downloadingFiles.Count == existingDataFiles && !Program.FunctionalTest) return;
+
+            _logger.Log(string.Format(Resources.ConfigRunner_DownloadData_Found__0__matching_data_files_on__1__, downloadingFiles.Count, server.GetUrl()));
+            foreach (var file in downloadingFiles)
+                _logger.Log(file);
+            _logger.Log(Resources.ConfigRunner_DownloadData_Starting_download___);
+            
+            var ftpClient = server.GetFtpClient();
+            var source = new CancellationTokenSource();
+            CancellationToken token = source.Token;
+            _runningCancellationTokens.Add(source);
+            var i = 0;
+            int triesOnFile = 0;
+            while ( i < downloadingFiles.Count && IsRunning())
+            {
+                var fileName = downloadingFiles[i];
+                if (triesOnFile == 0)
+                {
+                    _logger.Log(string.Format(Resources.ConfigRunner_DownloadData__0____1__of__2__, fileName, i + 1, downloadingFiles.Count));
+                }
+                
+                var localFileDestination = Path.Combine(mainSettings.DataFolderPath, fileName);
+                var currentFileSize = (double) allFiles[fileName].Size;
+
+                if (File.Exists(localFileDestination) && Math.Abs(new FileInfo(localFileDestination).Length - currentFileSize) < 0.0001)

```

Comment:
Why are you comparing these values as doubles instead of long integers?

Reply:
I forgot I casted to the double, oops